crates/forge_analyzer/src/definitions.rs

```diff
@@ -1193,16 +1203,19 @@ impl<'cx> FunctionAnalyzer<'cx> {
             }
         }
         if let Expr::Member(MemberExpr { obj, prop, .. }) = expr {
-            if let Some(class) = self.resolve_class_prop(obj) {
-                if let MemberProp::Ident(ident) = prop {
-                    if let Some((_, def_constructor)) = class
-                        .pub_members
-                        .iter()
-                        .find(|(name, defid)| name == &ident.sym)
+            if let MemberProp::Ident(ident) = prop {
```

Comment:
Looking at `resolve_class_prop` and `resolve_object_prop`, you should be able to accept `Ident` instead of `Expr` and just push the check to the caller. i.e. also check if `obj` is an `Expr::Ident`, and then pass the `ident` to both functions.

Reply:
I am now passing in the ident instead. I am slightly confused by pushing the check to the caller. The check to see if there is a property in the class? Sorry 😅

Reply:
Ah, the `get_prop_defid_from_object` method takes a `MemberProp`, however, it returns `None` if the `MemberProp` isn't an `Ident`. Furthermore, you already destructure the `MemberProp::Ident` variant before, the call, so you can just accept an `Ident` instead of `MemberProp`.

That way, if a caller already has an `Ident`, then you don't need to wrap it in a `MemberProp`, and it's more transparent on what's needed to get a property. This [article](https://matklad.github.io/2023/11/15/push-ifs-up-and-fors-down.html#Push-Ifs-Up) explains the concept pretty well.

Reply:
Thank you. That article was very helpful. I do not require that check since it is already in the function. Resolved. Thanks again.